### ISSUE TITLE
pm2 module does not detect running node

### ISSUE THREAD

**Issue by CodingJonas** [cubedro/eth-netstats#316]
I want to display statistics about a private network, but it seems the `eth-net-intelligence-api` can't detect the running node I want to display.

This is how I start the node I want to display in netstats:
```
#!/bin/bash

geth --identity "miner1" \
    --networkid 42 \
    --datadir "~/Blockchain/private_geth_network/miner1" \
    --nodiscover \
    --mine \
    --rpc \
    --rpcport "8051" \
    --port "30351" \
    --unlock 0 \
    --password "/Users/jonas/Blockchain/private_geth_network/miner1/password.sec" \
    --ipcpath "~/Library/Ethereum/geth.ipc" \
    console
````

And this is the `app.json` file I use to run `pm2 start app.sjon`:
```
[
  {
    "name"              : "miner1",
    "cwd"               : ".",
    "script"            : "app.js",
    "log_date_format"   : "YYYY-MM-DD HH:mm Z",
    "merge_logs"        : false,
    "watch"             : false,
    "max_restarts"      : 10,
    "exec_interpreter"  : "node",
    "exec_mode"         : "fork_mode",
    "env":
    {
      "NODE_ENV"        : "production",
      "RPC_HOST"        : "localhost",
      "RPC_PORT"        : "8051",
      "LISTENING_PORT"  : "30351",
      "INSTANCE_NAME"   : "miner1-1",
      "CONTACT_DETAILS" : "",
      "WS_SERVER"       : "http://localhost:3000",
      "WS_SECRET"       : "hello",
      "VERBOSITY"       : 2
    }
  }
]
```

But `pm2` gives me this message:
```
[PM2] Spawning PM2 daemon with pm2_home=/Users/jonas/.pm2
[PM2] PM2 Successfully daemonized
[PM2][WARN] Applications miner0 not running, starting...
[PM2] App [miner0] launched (1 instances)
┌──────────┬────┬──────┬───────┬────────┬─────────┬────────┬─────┬───────────┬──────┬──────────┐
│ App name │ id │ mode │ pid   │ status │ restart │ uptime │ cpu │ mem       │ user │ watching │
├──────────┼────┼──────┼───────┼────────┼─────────┼────────┼─────┼───────────┼──────┼──────────┤
│ miner0   │ 0  │ fork │ 12230 │ online │ 0       │ 0s     │ 0%  │ 12.7 MB   │ jonas│ disabled │
└──────────┴────┴──────┴───────┴────────┴─────────┴────────┴─────┴───────────┴──────┴──────────┘
```

Can you see the problem?

**Comment by mikeyb** [comment]
Would need to see the error log to see the problem.  You might try reading the pm2 docs to understand the software you are attempting to utilize

http://pm2.keymetrics.io/docs/usage/quick-start/

This Issue tracker is for bugs, not help desk requests

**Comment by CodingJonas** [comment]
Sorry mikeyb, I tried every possible way to set everything up as discribed in the different wiki and READMEs around, but couldn't resolve this.

It appears the problem is with pm2 not properly connecting:
```
2018-02-22 18:48 +01:00: [wsc] =✘= Socket error: { Error: connect ECONNREFUSED 127.0.0.1:3000
    at Object._errnoException (util.js:1003:13)
    at _exceptionWithHostPort (util.js:1024:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1195:14)
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 3000,
  type: 'error',
  target: 
   WebSocket {
     domain: null,
     _events: { error: [Function: onerror] },
     _eventsCount: 1,
     _maxListeners: undefined,
     _socket: null,
     _ultron: null,
     _closeReceived: false,
     bytesReceived: 0,
     readyState: 3,
     supports: { binary: true },
     extensions: null,
     _binaryType: 'arraybuffer',
     _isServer: false,
     url: 'ws://localhost:3000/api/?_primuscb=M6-eNQn',
     protocolVersion: 13,
     _closeTimer: null,
     _closeCode: 1006 } }
```

I will look into it further. Still some help would be awesome. Maybe more user trying to set netstats up have this problem, so it would be worth discussing it here and fix the issue behind it :)

**Comment by mikeyb** [comment]
You might check closed issues, but this error is stating that you don't have anything running on port 3000 via localhost.  So some dependency is not running, or there is a firewall blocking access to localhost port 3000

**Comment by CodingJonas** [comment]
Thanks for the advice. I did not know pm2 that well. Apparently this package expects an open port to connect to. I thought it would open the port itself.
I resolved this issue by running `eth-netstats` before starting the `eth-net-intelligence-api`.

I guess everyone with more experience would have known this. But maybe there should be a hint in the README or wiki, stating in which order to execute the different parts?
